extensions/2.0/Khronos/KHR_meshopt_compression/README.md

```diff
@@ -0,0 +1,675 @@
+
+**Channel 1 (2-byte deltas)**: 2-byte deltas are computed as zigzag-encoded differences between 16-bit values of the element and the previous element in the same position; the zigzag encoding scheme works as follows:
+
+```
+encode(uint16_t v) = ((v & 0x8000) != 0) ? ~(v << 1) : (v << 1)
+decode(uint16_t v) = ((v & 1) != 0) ? ~(v >> 1) : (v >> 1)
+```
+
+The deltas are computed in 16-bit integer space with wrap-around two-complement arithmetic. Values are assumed to be little-endian, so the least significant byte is encoded before the most significant byte.
+
+**Channel 2 (4-byte XOR deltas)**: 4-byte deltas are computed as XOR between 32-bit values of the element and the previous element in the same position, with an additional rotation applied based on the high 4 bits of the channel mode byte:
+
+```
+rotate(uint32_t v, int r) = (v << r) | (v >> (32 - r))
+```
+
+The deltas are computed in 32-bit integer space. Values are assumed to be little-endian, so the least significant byte is encoded before the most significant byte.
+
+Because the channel mode defines encoding for 4 bytes at once, it's impossible to mix modes 0 and 1 within the same channel: if the first 2-byte group of an aligned 4-byte group uses 2-byte deltas, the second 2-byte group must use 2-byte deltas as well.
+
+## Mode 1: triangles
+
+Triangle compression compresses triangle list indices by exploiting similarity between consecutive triangles. Given a triangle stream that has been optimized for locality, very often subsequent triangles share an edge with the recently encoded triangle. The encoder uses a few other techniques to try to encode most triangles in optimized triangle lists into a single byte.
+
+The encoded stream structure is as follows:
+
+- Header byte, which must be equal to `0xe1`
+- Triangle codes, referred to as `code` below, with a single byte for each triangle
+- Extra data which is necessary to decode triangles that don't fit into a single byte, referred to as `data` below
+- Tail block, which consists of a 16-byte lookup table, referred to as `codeaux` below
+
+Note that there is no way to calculate the length of a stream; instead, it is expected that the input stream is correctly sized (using `byteLength`) so that the tail block element can be found.
+
+There are two limitations on the structure of the 16-byte lookup table:
+
+- The last two bytes must be 0
+- The remaining bytes must not contain any nibbles equal to `0xf`.
+
+During the decoding process, decoder maintains four variables:
+
+- `next`: an integer referring to the expected next unique index (also known as high-watermark), starts at 0
+- `last`: an integer referring to the last encoded index, starts at 0
+- `edgefifo`: a 16-entry FIFO with two vertex indices in each entry; initial contents is undefined
+- `vertexfifo`: a 16-entry FIFO with a vertex index in each entry; initial contents is undefined
```

Comment:
Please clarify data types of these variables (looks like unsigned 32-bit integers).